src/adapter/src/catalog/builtin.rs

```diff
@@ -2371,6 +2371,67 @@ JOIN mz_catalog.mz_roles role_owner ON role_owner.id = class_objects.owner_id
 WHERE mz_schemas.database_id IS NULL OR d.name = pg_catalog.current_database()",
 };
 
+pub const PG_DEPEND: BuiltinView = BuiltinView {
+    name: "pg_depend",
+    schema: PG_CATALOG_SCHEMA,
+    sql: "CREATE VIEW pg_catalog.pg_depend AS WITH class_objects AS (
+    SELECT 
+        id,
+        CASE
+            WHEN type = 'table'  THEN 'pg_tables'::pg_catalog.regclass::pg_catalog.oid
+            WHEN type = 'source' THEN 'pg_tables'::pg_catalog.regclass::pg_catalog.oid
+            WHEN type = 'view'   THEN 'pg_views'::pg_catalog.regclass::pg_catalog.oid
+            WHEN type = 'materialized-view' THEN 'pg_matviews'::pg_catalog.regclass::pg_catalog.oid
+        END classid,
+        schema_id
+    FROM mz_catalog.mz_relations
+    UNION ALL
+    SELECT mz_indexes.id, mz_indexes.oid, schema_id
+    FROM mz_catalog.mz_indexes
+    JOIN mz_catalog.mz_relations ON mz_indexes.on_id = mz_relations.id
+),
+
+current_objects AS (
+    SELECT 
+        class_objects.id,
+        classid,
+        mz_objects.oid,
+        CASE
+            WHEN mz_catalog.mz_schemas.name = 'pg_catalog' THEN 'i' 
+            WHEN mz_catalog.mz_schemas.name = 'mz_catalog' THEN 'i' 
+            WHEN mz_catalog.mz_schemas.name = 'mz_internal' THEN 'i'
+            -- We mark sources as internal because they
+            -- are exposed via pg_catalog
+            -- as tables, but the system will error
+            -- if a user attempts DROP TABLE <source>.
+            WHEN (
+                SELECT count(*) FROM mz_catalog.mz_sources 
+                WHERE id = class_objects.id
+            ) > 0 THEN 'i'
```

Comment:
Rather than re-deriving the type with a subquery it will be more efficient to plumb the `type` column through from `class_objects`. (But also this bit may just go away; see next comment.)